src/commands/removeCodeBlocks.ts

```diff
@@ -0,0 +1,470 @@
+import { CommandModule } from "yargs";
+import { promises as fs } from "fs";
+import MagicString from "magic-string";
+import { visit } from "../tree";
+import restructured from "../restructured";
+import { CanonicalLanguageValues } from "../types/CanonicalLanguageValues";
+import { LanguageFileExtensions } from "../types/LanguageFileExtensions";
+import { LanguageValueMappings } from "../types/LanguageValueMappings";
+import * as path from "path";
+import { LanguageMapper } from "../types/LanguageMapper";
+import { CodeBlockWithMetadata } from "../types/CodeBlockWithMetadata";
+import { PageWriteData } from "../types/PageWriteData";
+
+export const removeCodeBlocks = async (
+  pageFilepath: string,
+  absFilepathToSource: string,
+  source: string
+): Promise<MagicString> => {
+  const document = new MagicString(source);
+  const rst = restructured.parse(document.original, {
+    blanklines: true,
+    indent: true,
+    position: true,
+  });
+
+  const codeBlockDirectory =
+    makeCodeBlockDirectoryFromPageFilepath(pageFilepath);
+  const codeBlocks: CodeBlockWithMetadata[] = [];
+  let codeBlockInstance = 0;
+  try {
+    visit(rst, (node) => {
+      if (node.children === undefined) {
+        return;
+      }
+      if (node.type !== "directive") {
+        return;
+      }
+      if (node["directive"] !== "code-block") {
+        return;
+      }
+
+      // We use the code block instance number on the page as a name for the code block file
+      codeBlockInstance++;
+
+      /* If a writer has provided a language for the code block, its value is the `args` field.
+       * If the writer has not provided a value, set the default lang value to 'text'.
+       */
+      let language = "text";
+      if (node.args !== undefined) {
+        language = node.args as string;
+      }
```

Comment:
```suggestion
      const language = node.args ?? "text";
```
Some say this is harder to read but I find the `let` distracting because I need to scan why it isn't a const.
You also don't need to cast as string once it's proven (via if or assert) to the compiler that it's not undefined.

Reply:
Makes sense. I tend to err on the side of verbose for clarity since a lot of my code is for docs consumption, but I think I've refactored this and all similar instances in across this PR.